### ISSUE TITLE
OpenAI or Azure OpenAI API key not found (Cannot set apiKey - ENV works fine)

### ISSUE THREAD

**Issue by scottforte** [langchain-ai/langchainjs#5713]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain.js documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain.js rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

```js
const { OpenAI } = require("@langchain/openai");

// WORKS FINE
new OpenAI({
  model,
  temperature,
});

// FAILS
new OpenAI({
  model,
  temperature,
  apiKey: `same exact key - ENV removed - yes, confirmed it's there in the code via console before calling`,
});

```

### Error Message and Stack Trace (if applicable)

Error: OpenAI or Azure OpenAI API key not found\n' +
        '    at new OpenAIChat ([redacted]\\node_modules\\@langchain\\openai\\dist\\legacy.cjs:235:19)\n' +
        '    at new OpenAI ([redacted]\\node_modules\\@langchain\\openai\\dist\\llms.cjs:79:20)\n' +

rest redacted

### Description

Setting the ENV: OPENAI_API_KEY the creating an model works fine, but passing a string fails. 

This is only for `new OpenAI()` 

`new OpenAIEmbeddings()` works fine to pass an apiKey string in

### System Info

Windows
NPM 10.5.0
Node v20.12.2

**Comment by scottforte** [comment]
Problem is here:
api\node_modules\@langchain\openai\dist\legacy.cjs:230

fields?.openAIApiKey ?? (0, env_1.getEnvironmentVariable)("OPENAI_API_KEY");
should be: fields?.apiKey ?? (0, env_1.getEnvironmentVariable)("OPENAI_API_KEY");

I'll try to add a PR

**Comment by jacoblee93** [comment]
Merged a fix in #5714, but also these are legacy classes that generally shouldn't be used.
